### DISCUSSION TITLE
End of Speech Detection

### DISCUSSION THREAD

**Discussion by atul-dorge** [deepgram/community#910] [General help]
Context: We have an app which creates x (x = 5-10) concurrent websocket connections to Deepgram's Real time Speech to Text Service. The app sends audio and expects text transcriptions in real time. 

Problem: We are receiving the text transcriptions after a lag of 2-3 seconds. I.e the time difference between the time of last audio chunk we sent to the time at which we receive transcript with speech_final = True is as high as 2-3 seconds. 

This doesn't serve it to be real time. Further we expect the "Utterance End" to be received as soon as the last audio chunk with voice is sent. We are receiving his with 2-3 seconds lag as well.

How do we try resolving above 2 issues?

Full Deepgram URL: wss://api.deepgram.com/v1/listen?punctuate=false&model=nova-2&sample_rate=8000&channels=1&encoding=linear16&language=hi&interim_results=true&utterance_end_ms=1000&vad_events=true&endpointing=110

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.
